en/scripting/modules/example.md

```diff
@@ -0,0 +1,248 @@
+// Using the light version
+import protobuf from 'protobufjs/light.js';
+// Using the minimal version
+import protobuf from 'protobufjs/minimal.js';
+```
+
+> **Note**: in the case of protobufjs and many classic npm packages, the suffix is required when importing subpaths in the package. Please refer to the [Cocos Creator module specification](./spec.md).
+
+#### TypeScript configuration
+
+When importing modules without default export in TypeScript, it usually appears that `Module '"/${project_path}/protobufjs"' has no default export.`. This is because currently `protobufjs` only provides CommonJS modules, and Cocos Creator accesses CommonJS modules via "default import", but CommonJS modules really don't have a "default export". In this case, you can set the `"allowSyntheticDefaultImports"` option in the `"compilerOptions"` field to `true` by editing the `tsconfig.json` file in the project directory. If the field is not present, you can fill it in yourself.
+
+```json5
+{
+  /* Base configuration. do not edit this field. */
+  "extends": "./temp/tsconfig.cocos.json",
+
+  "compilerOptions": {
+      "allowSyntheticDefaultImports": true, // Needs to be turned on
+  }
+}
+```
+
+### Case 2: Compiling a proto file from protobuf into a JavaScript file
+
+This section is about how to compile a proto file into a JavaScript file. In fact, when looking through the protobufjs documentation, you can see that it provides [command line tools](https://www.npmjs.com/package/protobufjs#command-line) to convert static modules and ts declaration files. This time, let's take an empty 3.0 project, example, as an example to demonstrate the whole process.
+
+First, install `protobufjs` via npm and write it to the `package.json` dependency in the project directory.
+
+Next, create a new Proto directory in the project directory and define a few proto files.
+
+```proto
+// pkg1.proto
+
+package pkg1;
+syntax = "proto2";
+message Bar {
+    required int32 bar = 1;
+}
+
+// pkg2.proto
+
+package pkg2;
+syntax = "proto2";
+message Baz {
+	required int32 baz = 1;
+}
+
+// unpkg.proto is not part of any package
+
+syntax = "proto2";
+message Foo {
+    required int32 foo = 1;
+}
+```
+
+Next, define it in package.json.
+
+```json
+"scripts": {
+    "build-proto:pbjs": "pbjs --dependency protobufjs/minimal.js --target static-module --wrap commonjs --out ./Proto.js/proto.js ./Proto/*.proto",
+    "build-proto:pbts": "pbts --main --out ./Proto.js/proto.d.ts ./Proto.js/*.js"
+},
+```
+
+The first directive `build-proto:pbjs` roughly means to compile the proto file into js. The parameter `--dependency protobufjs/minimal.js` is actually because the execution `require`s to `protobufjs`, but all we need to use is its submodule `minimal.js`. Then, generate the js into the Proto.js folder (note: if you don't have a Proto.js folder, you need to create it manually). The second command `build-proto:pbts` generates the type declaration file based on the output of the first paragraph.
```

Comment:
```suggestion
The first directive `build-proto:pbjs` roughly means to compile the proto file into js. The parameter `--dependency protobufjs/minimal.js` is actually because the execution `require`s to `protobufjs`, but all we need to use is its submodule `minimal.js`. Then, generate the js into the Proto.js folder.

> **Note**: if a Proto.js folder is not present, manually create it. 

The second command `build-proto:pbts` generates the type declaration file based on the output of the first paragraph.
```